docs/design-rate-limit-otp-hardening.md

```
@@ -0,0 +1,625 @@
+
+- 不重写 OAuth 登录链路。
+- 不引入新的 Redis 客户端或新的基础设施（继续复用 `redis.Client` 全局实例与现有 `Set/Get/Del/Exists/Incr` 封装）。
+- 不在本期实现「一次性 verify token 绑定」（见 §13 未来演进，列为后续可选项）。
+
+---
+
+## 2. 现状梳理（实现基线）
+
+### 2.1 三步注册流程
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)